paddle isnan isinf isfinit

Crayon-ShinChan opened the pull request.

ivy-leaves commented:
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

Crayon-ShinChan commented:
> If you are working on an open task, please **edit** the PR description to link to the issue you've created.
> 
> For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).
> 
> Thank you 🤗

Hi,

I tried to search how to link a PR to issues from https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#

But I didn't see the setting icon on my page? Can I ask how to link the PR to my issues?

<img width="1301" alt="image" src="https://github.com/unifyai/ivy/assets/38809606/a227a228-8746-4b0f-ab43-40a1374c9f20">

Crayon-ShinChan commented:
And the test fail is because `amax` function which committed here https://github.com/unifyai/ivy/commit/c69258d556492d7d4c8c3dc9be531a4111d3857c

<img width="1227" alt="image" src="https://github.com/unifyai/ivy/assets/38809606/4bceb06a-1412-4e02-a22e-f9f6ee57cfdc">

Can I ask that if there is anything I should do for it?

KareemMAX commented:
> But I didn't see the "setting" gear icon on my page? Can I ask how to link the PR to my issues?

That's because you're not a member of our organization, you can instead write `Close <issue link>` in the issue body as mentioned in our contribution docs.

PS: I did that for you

KareemMAX commented:
> And the test fail is because `amax` function which committed here [c69258d](https://github.com/unifyai/ivy/commit/c69258d556492d7d4c8c3dc9be531a4111d3857c)

It seems like this commit is not related to your work, as long as **your code** is passing tests we will merge it

KareemMAX commented:
To streamline the review process a little bit, would you mind making a comment with `add_frontend_checklist` in it?
This will generate a checklist for you that you can check which parts have you completed

Crayon-ShinChan commented:
`add_frontend_checklist`

KareemMAX commented:
@Crayon-ShinChan yup, just without the having it wrapped as code 😅. I've edited your comment but sadly it won't trigger our bot to fill the checklist. Can do you do it once again? But just "add_frontend_checklist" without the double quotes

Crayon-ShinChan commented:
add_frontend_checklist

Crayon-ShinChan commented:
> @Crayon-ShinChan yup, just without the having it wrapped as code 😅. I've edited your comment but sadly it won't trigger our bot to fill the checklist. Can do you do it once again? But just "add_frontend_checklist" without the double quotes

done! thank you for the instruction!

KareemMAX commented:
Please check on anything you think you have completed (or should skip), and I'll review it accordingly. Sorry for the vague instructions at first 😅 I should have told you that earlier

Crayon-ShinChan commented:
> Please check on anything you think you have completed (or should skip), and I'll review it accordingly. Sorry for the vague instructions at first 😅 I should have told you that earlier

Hi, thank you for your instruction~ I checked the bullet points that I'm really sure but for some that I'm not certainly sure, I left them there.

For checks run on Github. I saw many new PRs meet the error in `run_tests (1)` which the error message is
```
_____ test_paddle_allclose[cpu-ivy.functional.backends.torch-False-False] ______
ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:962: in test_paddle_allclose
    class_tree=CLASS_TREE,
ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py:490: in dtype_and_values
    available_dtypes = draw(available_dtypes)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1563: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:173: in get_dtypes
    valid_dtypes = set(retrieval_fn(test_globals.CURRENT_BACKEND(), kind))
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:35: in _get_fn_dtypes
    return test_globals.CURRENT_RUNNING_TEST.supported_device_dtypes[framework.backend][
E   KeyError: 'cpu'
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.79.2', b'AA==') as a decorator on your test case
```

Is there anything I can do for my PR now? Please feel free to tell me, I'm pretty new here.

KareemMAX commented:
@Crayon-ShinChan Don't worry about the CI tests, we double check on your frontend function as well

KareemMAX reviewed (changes requested):
Awesome work @Crayon-ShinChan 🚀🚀

Your PR is almost ready to merge but few changes need to be made first. Please look into the points I left them with the ❌ mark.

The main issues are:
- The use of `get_dtypes("float")` instead of `get_dtypes("valid")
- `.idea` folder shouldn't be changed

Please once you finish making those changes request a re-review

Crayon-ShinChan commented:
> Awesome work @Crayon-ShinChan 🚀🚀
> 
> Your PR is almost ready to merge but few changes need to be made first. Please look into the points I left them with the ❌ mark.
> 
> The main issues are:
> 
> * The use of `get_dtypes("float")` instead of `get_dtypes("valid")
> * `.idea` folder shouldn't be changed
> 
> Please once you finish making those changes request a re-review

Hi Kareem, I've fixed the two issues. Thank you for your time to review my code.

I've got one question which is about .idea folder. I saw `.idea/*` in `.gitignore` file and I thought my changes in `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` won't be committed. However, when I changed these three files, they were still committed. Then I have to use `git update-index --assume-unchanged .idea/ivy.iml .idea/misc.xml .idea/vcs.xml` to ask git to ignore these files explicitly without changing `.gitignore` file.

Can I ask why `.idea/*` in `.gitignore` but my changes on `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` will still be committed? (I have to change `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` in my local M1 MacBook because I can't use remote docker and I used local conda environment)

Crayon-ShinChan commented:
> Awesome work @Crayon-ShinChan 🚀🚀
> 
> Your PR is almost ready to merge but few changes need to be made first. Please look into the points I left them with the ❌ mark.
> 
> The main issues are:
> 
> * The use of `get_dtypes("float")` instead of `get_dtypes("valid")
> * `.idea` folder shouldn't be changed
> 
> Please once you finish making those changes request a re-review

@KareemMAX Hi Kareem, I've fixed both issues you mentioned. Could you please review my PR? Thank you for your time!

The last commit meet error of "check formatting" is because two other people declare the same function, I'm not sure whether should I fix it so I just leave it there.

<img width="618" alt="image" src="https://github.com/unifyai/ivy/assets/38809606/42de96ad-fff4-4aef-aebe-c8bcde9a5fe9">
<img width="680" alt="image" src="https://github.com/unifyai/ivy/assets/38809606/cafef9c1-fe51-47eb-8115-f3a7c23b1a11">

KareemMAX commented:
> Hi Kareem, I've fixed the two issues. Thank you for your time to review my code.
> 
> I've got one question which is about .idea folder. I saw `.idea/*` in `.gitignore` file and I thought my changes in `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` won't be committed. However, when I changed these three files, they were still committed. Then I have to use `git update-index --assume-unchanged .idea/ivy.iml .idea/misc.xml .idea/vcs.xml` to ask git to ignore these files explicitly without changing `.gitignore` file.
> 
> Can I ask why even we have `.idea/*` in `.gitignore`, my changes on `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` will still be committed? (I have to change `.idea/ivy.iml .idea/misc.xml .idea/vcs.xml` in my local M1 MacBook because I can't use remote docker and I used local conda environment)

This is actually something about how git works, if we have a file in our tree even if it is ignored. git will still track it. You should use commands to make git not track it as you did sadly 😅. I usually don't stage it at all

KareemMAX commented:
> @KareemMAX Hi Kareem, I've fixed both issues you mentioned. Could you please review my PR? Thank you for your time!
> 
> The last commit meet error of "check formatting" because two other people declare the same function, I'm not sure whether I should fix it, so I just leave it there.

Oh, that actually qualify your PR as approved then. Please fix the merge conflicts and request a review from me and I'll merge your PR right away

KareemMAX reviewed (changes requested):
LGTM! 🚀🚀 Just resolve merge conflicts and we will be ready to merge
